#164139679 Set up swagger for API documentation

codinger41 opened the pull request:
#### What does this PR do?
It sets up swagger for API documentation.

#### Description of Task to be completed?
It sets up swagger for API documentation and renders the UI on `${HOST_URL}/api-docs`

#### How should this be manually tested?
- Run `npm run start:dev` to start the application.
- Visit `${HOST_URL}/api-docs` to view the documentation UI.

#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
#164139679
#### Screenshots (if appropriate)
<img width="1280" alt="screenshot 2019-03-04 at 2 01 44 pm 1" src="https://user-images.githubusercontent.com/26174035/53735061-13c92580-3e86-11e9-95fd-ae3c7ffdb7d7.png">

#### Questions:
N/A

codinger41 committed:
chore(documentation): setup swagger for api docs

- add doc.json file with swagger configuration
- add swagger-ui-express to render documentation

[Finishes #164139679]

youngestdj commented on doc.json:
```diff
@@ -0,0 +1,54 @@
+      "url": "https://www.gnu.org/licenses/gpl-3.0.en.html"
+    }
+  },
+  "host": "authors-haven-staging.herokuapp.com",
```
URL should be repo-name-staging.herokuapp.com for staging and repo-name.herokuapp.com for production.

youngestdj replied:
```suggestion
  "host": "vidar-ah-backend-staging.herokuapp.com",
```

codinger41 committed:
chore(heroku) deploy to heroku
- change sequelize migration script to use sequelize from node_modules

[#Finishes #164139676]

codinger41 committed:
chore(heroku) deploy to heroku
- change sequelize migration script to use sequelize from node_modules

[#Finishes #164139676]

codinger41 committed:
chore(heroku) deploy to heroku
- change sequelize migration script to use sequelize from node_modules

[#Finishes #164139676]

codinger41 committed:
chore(heroku) deploy to heroku
- change sequelize migration script to use sequelize from node_modules

[#Finishes #164139676]

codinger41 committed:
chore(heroku) deploy to heroku
- change sequelize migration script to use sequelize from node_modules

[#Finishes #164139676]

codinger41 committed:
chore(documentation): setup swagger for api docs

- add doc.json file with swagger configuration
- add swagger-ui-express to render documentation

[Finishes #164139679]

9jaswag commented on package.json:
```diff
@@ -5,16 +5,22 @@
   "scripts": {
     "coverage": "npm run db-migrate && mocha --reporter mochawesome --timeout=20000 --require babel-register --exit",
-    "db-migrate": "sequelize db:migrate:undo:all && sequelize db:migrate",
+    "db-migrate": "node_modules/.bin/sequelize db:migrate:undo:all && sequelize db:migrate",
```
this should not be referenced from node modules. The sequelize cli in dependencies should handle this.

9jaswag replied:
Also, in production, we don't want to drop and run all migrations whenever we deploy to production. What happens to our data then? There's a way to only run migrations only when there's a migration to be run. Find out how. @leksyib

codinger41 replied:
Thanks, @9jaswag  will do.

9jaswag commented on package.json:
```diff
@@ -5,16 +5,22 @@
     "lint": "eslint routes models config test index.js",
     "test": "npm run db-migrate && mocha test/index.js --timeout=20000 --require babel-register --exit",
-    "start": "sequelize db:migrate && babel-node index.js",
+    "start": "node_modules/.bin/sequelize db:migrate && babel-node index.js",
```
babel-node should not be used in production. The code should be "transpiled" and run by Node

9jaswag commented on .gitignore:
```diff
@@ -42,3 +42,4 @@ mochawesome-report
 
 # Ignore yarn generated files
 yarn.lock
+dist/
```
fix EOL issue

codinger41 replied:
Fixed
